src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -189,20 +185,42 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
 
         // Keep track of a token per document so that we can cancel the search if the doc is closed
         const preferredSearchToken = new CancellationTokenSource();
-        this.findPreferredInProgress.set(document, preferredSearchToken);
```

Comment:
Refactored this as I felt we had a few too many methods/properties here.
Removed a private property & three methods.
I think all of them fit in a single method. Not sure this improves readability, if not please feel free to reject this PR.
It took me some time to move into findPreferredInProgress & see where that is used, & then 
serach & look at findPreferredKernel & finally setPreferredController too.

Reply:
Again, its possible , the readability is worse, if thats the case, please do let me know.